orttraining/orttraining/python/training/ortmodule/_options.py

```diff
@@ -0,0 +1,336 @@
+
+    def _validate(self, log_level):
+        # check if log_level is an instance of LogLevel
+        if not isinstance(log_level, LogLevel):
+            raise TypeError(f"Expected log_level of type LogLevel, got {type(log_level)}.")
+
+    @property
+    def log_level(self) -> LogLevel:
+        return self._log_level
+
+
+class DebugOptions:
+    """Configurable debugging options for ORTModule.
+
+    Args:
+        log_level (:obj:`LogLevel`, optional): Configure ORTModule log level. Defaults to LogLevel.WARNING.
+            log_level can also be set by setting the environment variable "ORTMODULE_LOG_LEVEL" to one of
+            "VERBOSE", "INFO", "WARNING", "ERROR", "FATAL". In case both are set, the environment variable
+            takes precedence.
+        save_onnx (:obj:`bool`, optional): Configure ORTModule to save onnx models. Defaults to False.
+            The output directory of the onnx models by default is set to the current working directory.
+            To change the output directory, the environment variable "ORTMODULE_SAVE_ONNX_PATH" can be
+            set to the destination directory path.
+        onnx_prefix (:obj:`str`, optional): Name prefix to the ORTModule ONNX models saved file names.
+            Must be provided if save_onnx is True
+
+    Raises:
+        OSError: If save_onnx is True and output directory is not writable.
+        TypeError: If save_onnx is True and name_prefix is not a valid string. Or if
+            log_level is not an instance of LogLevel.
+        ValueError: If save_onnx is True and name_prefix is an empty string.
+
+    """
+
+    def __init__(self, log_level=LogLevel.WARNING, save_onnx=False, onnx_prefix="", save_path="", config=None):
+        self.log_level = log_level
+        self.save_onnx = save_onnx
+        self.onnx_prefix = onnx_prefix
+
+        self._save_onnx_models = _SaveOnnxOptions(self.save_onnx, self.onnx_prefix, save_path)
+        self._logging = _LoggingOptions(self.log_level)
+
+    @property
+    def save_onnx_models(self):
+        """Accessor for the ONNX saving configuration."""
+
+        return self._save_onnx_models
+
+    @property
+    def logging(self):
+        """Accessor for the logging configuration."""
+
+        return self._logging
+
+
+class _SkipCheck(IntFlag):
+    """Enumeration to specify which checks should be skipped, allowing faster execution"""
+
+    SKIP_CHECK_DISABLED = 1
+    SKIP_CHECK_DEVICE = 2
+    SKIP_CHECK_BUILD_GRADIENT = 4
+    SKIP_CHECK_EXECUTION_AGENT = 8
+
+    def is_set(self, check):
+        """Check whether `check` is set on the `_SkipCheck instance
+
+        SKIP_CHECK_DISABLED implies the check will return False
+        """
+
+        return not _SkipCheck.is_disabled(self) and check in self
+
+    def is_disabled(self):
+        """Check whether `_SkipCheck.SKIP_CHECK_DISABLED is set on the `_SkipCheck instance"""
+
+        return _SkipCheck.SKIP_CHECK_DISABLED in self
+
+
+class _RuntimeOptions:
```

Comment:
`RuntimeOptions` seems too broad a scope. It seems we have put all the configuration in this class? Do you think we could bucket similar configs into a group and then create a class for each of them?
They all can be a part of a larger `Options` class (including `DebugOptions`).